### DISCUSSION TITLE
How to leave Background color unchanged in TextBox:pointerover

### DISCUSSION THREAD

**Discussion by mkj-stonehill** [AvaloniaUI/Avalonia#12619] [Q&A] [answered]
I'm using Avalonia 0.10.18 (application runs on Raspberry Pi 4, Bullseye, in case that matters).  The application uses the Fluent style (and it's MVVM, also in case that matters).

I have one particular TextBox control whose Background and Foreground properties are bound to values, which the code will change in order to indicate different results.  However, if I move the mouse over this control, it changes the background color (which makes the white text unreadable).

Is there any way to define a Style for TextBox:pointerover that just leaves the background color alone?

**Comment by timunie** [comment]
look up original styles (mind the branch, you need to switch to `stable/0.10.x`):
https://github.com/AvaloniaUI/Avalonia/blob/88852695bcb519c5a13b01956df583920aaf992a/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml#L120-L132

**Reply by mkj-stonehill** [reply]
Thanks for the reply.  I did look at the Fluent styles (although, I confess, I forgot to change to the correct branch).  And, I can see how to override the background that the Fluent style sets (and I've done that, just to make sure I understood correctly).

However, what I need to do is have it not set the background at all during pointerover (leaving the background color that the code set through the bound property).  If I do not override the background color, I get whatever the Fluent style does.  Is there a magic value I can set the background to (in my style override) that will leave the background color alone?

**Reply by timunie** [reply, marked as answer]
No, just set it to exact same background color as normal state. In v11, this is possible but not in 0.10....

**Reply by mkj-stonehill** [reply]
*Smacks Head* I can't believe that did not occur to me!  Thank you!

Since I will probably be upgrading this project soon, how would one do that in v11?  I mean, setting it to the normal state would work, but what is new in v11 that is relevant here?

**Reply by timunie** [reply]
one can override this in 11.0: https://github.com/AvaloniaUI/Avalonia/blob/a18ad58547de8c8993cc171aa340fb3ace138d36/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml#L30

So you can swap out the entire theme for the button. Styling as done before still works :-)
